README.md

```diff
@@ -34,3 +34,205 @@ Read so far up to
 - Install [bun](https://bun.sh/)
 - Run `bun install` to install all dependencies
 - Run `bun test` to run all tests
+
+## Introduction to Cairo VM
+Cairo VM is an advanced virtual CPU architecture designed for blockchain technology and cryptographic operations. It's Turing-complete and STARK-friendly, efficiently handling complex program statements as polynomial equations over a finite field. This feature is crucial for secure and effective cryptographic proof management in blockchain applications, making Cairo VM a key player in the realm of decentralized computing.
+
+## Register-Based VM
+Cairo VM's register-based architecture allows for quicker data access compared to stack based interaction, enhancing computational efficiency and performance in cryptographic operations.
+
+## Detailed Architecture Overview
+- **Finite Field Operations**: Core to Cairo VM's computational power for cryptographic aspects of STARKs.
+
+- **Three-Register System**: The Cairo VM employs a three-register system, consisting of:
+
+    - **Program Counter (PC)**: Manages the sequence of program execution by holding the memory address of the next instruction.
+    - **Allocation Pointer (AP)**: Crucial for memory management, it points to the first unused memory cell, dynamically adjusting as the program runs.
+    - **Frame Pointer (FP)**: Facilitates access to a function's local memory, crucial for handling function calls and maintaining the state of local variables.
+
+## Advanced Functionality and Impact
+- **Scalability and Efficiency**: Integration with STARKs for efficient verification of computational claims.
+- **Security Enhancements**: Read-only memory model and finite field operations fortify Cairo VM against common vulnerabilities.
+- **Flexibility and Turing Completeness**: Suitable for a variety of applications, from simple smart contracts to complex decentralized applications.
+
+
+## Overall Cairo VM TS Flow
```

Comment:
is this automatically generated? not sure it's accurate anymore, and most probably subject to change.

Also not a fan of big graph that don't really bring any value to me as they are hard to read